### DISCUSSION TITLE
Required CODEOWNER pull requests and notifications

### DISCUSSION THREAD

**Discussion by AButler** [community/community#22479] [New to GitHub]
<p>We have recently enabled required CODEOWNER reviews for pull requests in to our master branch from at least one developer on the team (we have a GitHub team set up to catch all files in the codeowners file). Example CODEOWNERS file is:</p>
<pre><code>* @org-name/dev-team
/path/specific-file.xml @org-name/dev-sub-team
</code></pre>
<p>We want to require at least one code review from a member of the dev team and this does work with this set up, but it has the downside that now that the whole team is automatically assigned to new pull requests, meaning that every member of the team gets notifications for every pull request. </p>
<p>Is there anyway to require an approval from a member of the team without automatically assigning the team to the pull request to prevent them getting the notifications?</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/abutler">@abutler</a>,</p>
<p>Thanks for being part of the GitHub Community Forum! I’ll help as best I can.</p>
<p>You should be able to define CODEOWNERS as teams _or _individuals. So, instead of <code>@org-name/dev-team</code> you’d use the person’s username. There’s full instructions here: <a href="https://help.github.com/articles/about-codeowners/" rel="nofollow noopener">https://help.github.com/articles/about-codeowners/</a>.</p>
<p>If I’m misunderstanding, please let me know and we’ll continue to help as much as possible.</p>
<p>Cheers!</p>

**Comment by AButler** [comment]
<p>Hi,</p>
<p>Thanks for the response. Unfortuately no, I don’t want to switch to assigning individuals. I would like to prevent the auto-assignment of the code review to the team whilst still requiring that a code-review by one of the team is complete before merging.</p>
<p>This will enforce the approval process by a member of the team but not assign the whole team to the pull request so the whole team doesn’t get every pull request notification emails. </p>

**Comment by lee-dohm** [comment, marked as answer]
<p>For customized workflows like this, you may want to check out <a href="https://probot.github.io/" rel="nofollow noopener">Probot</a>. The <a href="https://probot.github.io/apps/wip/" rel="nofollow noopener">Work in Progress Probot app</a> gives an example of how to create a bot that can block the merging of PRs based on custom logic.</p>

**Comment by ozonni** [comment]
<p>We also have this problem. We have a lot of repositories and now i’m spammed with tons of emails. There should be a way to disable automatic PR review request from code owners.</p>

**Comment by metoro** [comment]
<p>Agreed. We have the same issue. This is making what would otherwise be a terrific new feature (codeowners) into something problematic. You should separate these two features and not auto-assign codeowners to PRs.</p>

**Comment by aommm** [comment]
<p>Agree with this. My team is a CODEOWNER for a repo, and it would be nice to avoid the email spam that this results in for all team members.</p>

**Comment by itaytay** [comment]
<p>We use the option to assign based on round robin.<br>
And this notification option is great for handling the email issue:<br>
“If assigning team members, don’t notify the entire team.”</p>
<p>However, it doesn’t solve the filters problem. I want to be able to filter based on reviews that are assigned specifically to me, but I’m also getting PRs that are assigned to the codeowner team, making the filters almost useless to me as a reviewer.</p>
<p>Is there a way to filter as I described? Maybe I’m missing something</p>

**Comment by nata7che** [comment]
<p>Is there a solution where a repository has CODEOWNERS file without having each PR assigned to it causing all the team to be notified?</p>

**Comment by JamesYangLim** [comment]
We have a different work style which we post our PR into a slack channel for our team to review. CODEOWNER file enforce codeowner to review the PR which is great. But some codeowners are not interested in the PR at all and they are getting spam! Is there a way that we can just disable the email notification entirely?

**Reply by Hronom** [reply]
Same here, we need to make sure that specific set of people can approve PR, this is for what we use `CODEOWNERS`. We also need to not send notifications to all team members trough email

**Comment by news-manishpatel** [comment]
Same, feature is great but I don't want the emails it is too much. I still want emails when someone responds to my comment or someone pushes changes. 

Any option to enable this?

**Comment by pietrodn** [comment]
I asked GitHub Support for help and they suggested checking the first checkbox (**"Only notify requested team members"**) in the **team settings**, for the team mentioned in `CODEOWNERS`. You need to be a GitHub org admin to change the team settings.

That solved the issue of notifications being sent to the team members not selected by the round-robin. 🥳 

![image](https://github.com/user-attachments/assets/3225ab76-b115-4c50-9c89-81b3fa460357)

**Reply by venkywonka** [reply]
just to confirm, this also works when the branch protection rule of "Require pull request reviews from Code Owners" is enabled right?
My understanding is that, the above rule prevents github from dropping the team-assignment as reviewer for the PR, even though individuals have been auto-assigned from that same team as reviewer. 

In such a scenario, will the whole team be notified, or only the reviewer?
Thanks!
